examples/cloud-composer-datafusion-ingestion-templates/composer/dag-dependencies/read_params.py

```diff
@@ -0,0 +1,286 @@
+environment parameter, dag paramter and task parameter
+"""
+import json
+import logging
+import os
+from airflow.models import Variable
+from airflow import AirflowException
+
+class ParamReader:
+    """ParamReader class reads parameters from the 3 parameter files
+    function read_parameters:
+        Reads parameters from param files
+        and assigns the values to the class variables.
+        This is the main function
+        to be called whenever parameters need to be read.
+    """
+    env_param_file=""
+    gcp_project_id = ""
+    location = ""
+    env_name = ""
+    audit_dataset = ""
+    source_regex = ""
+    target_dataset = ""
+    data_bucket = ""
+    archival_bucket = ""
+    composer_bucket = ""
+    cdf_namespace = ""
+    cdf_instance_name = ""
+    param_file_path = ""
+    source_type = ""
+    dag_load_type = ""
+    sftptogcs_dest_path = ""
+    gcs_src_file_path = ""
+    gcs_archival_path = ""
+    metadata_path =""
+    sftptogcs_dest_path =""
+    dbconnection = ""
+    port = ""
+    username = ""
+    input_database = ""
+    insert_audit_cols = ""
+    cdf_pipeline_timeout = ""
+    update_metadata = ""
+    refresh_authorized_views=""
+
+    task_params=[]
+
+    audit_ins_template = ""
+    audit_upd_template = ""
+
+    default_delimiter = ","
+    delimiter = ""
+
+    def get(self, param):
+        return getattr(self, param, None)
+
+    def read_parameters(self,system,dag_param_file_nm,task_param_file_nm):
+        """
+        Reads parameters from param files
+        and assigns the values to the class variables.
+        This is the main function
+        to be called whenever parameters need to be read.
+        :param system(string):
+            run id of the DAG generated by Airflow
+        :param dag_param_file_nm(string):
+            name of dag param file
+        :param task_param_file_nm(string):
+            name of task param file
+        """
+        logging.debug("Starting to initialize paramters from files")
+        logging.debug(f"gcp project initialized to {self.gcp_project_id}")
+
+        ###################################
+        # Read airflow variable for env parameter file
+        try:
+            self.env_param_file = Variable.get("env_param")
+        except KeyError:
+            logging.error(
+                "Missing airflow variable env_param "
+                "for environment parameter file")
+            raise
+
+        ###################################
+        # Get parameters from parameter files
+        ###################################
+
+        # Read env parameter file and set parameter values
+        logging.debug(
+            f"Reading environment parameter file "
+            f"from : {self.env_param_file}")
+        try:
+            with open(self.env_param_file) as env_param_json:
+                env_param = json.load(env_param_json)
+
+                self.gcp_project_id = env_param["gcp_project_id"]
```

Comment:
this is incredibly repetitive and thus unmaintainable.
instead of hard coding every key explicitly in the code just do:
```python3
for param, value in env_param.items:
    setattr(self, param, value)
```

After this you can have a `validate_params` method that checks expected keys are present and values obey your logic.